dev/sparktestsupport/modules.py

```diff
@@ -368,6 +368,7 @@ def __hash__(self):
         "pyspark.sql.pandas.group_ops",
         "pyspark.sql.pandas.types",
         "pyspark.sql.pandas.serializers",
+        "pyspark.sql.pandas.typehints",
```

Comment:
It looks like this contains only internal components. It wouldn't hurt to add leading underscore (probably applies to modules included with [SPARK-30434]).

Reply:
Let me keep it consistent in the current codebase for now. We could do that separately since there are already smilar private modules without underscore.

Reply:
Makes sense, though we should probably keep that mind in the future. Latest refactoring of `pyspark.ml` moved things in that direction, so it would be nice to keep things consistent (maybe some official guidelines?).